include/aikido/planner/ompl/detail/OMPLConfigurationToConfigurationPlanner-impl.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef AIKIDO_PLANNER_OMPL_DETAIL_OMPLCONFIGURATIONTOCONFIGURATION_IMPL_HPP_
+#define AIKIDO_PLANNER_OMPL_DETAIL_OMPLCONFIGURATIONTOCONFIGURATION_IMPL_HPP_
+
+#include "aikido/planner/ompl/OMPLConfigurationToConfigurationPlanner.hpp"
+
+#include <utility>
+
+#include "aikido/constraint/TestableIntersection.hpp"
+#include "aikido/constraint/dart/FrameDifferentiable.hpp"
+#include "aikido/constraint/dart/FrameTestable.hpp"
+#include "aikido/constraint/dart/JointStateSpaceHelpers.hpp"
+#include "aikido/distance/defaults.hpp"
+#include "aikido/planner/ompl/GeometricStateSpace.hpp"
+#include "aikido/planner/ompl/MotionValidator.hpp"
+#include "aikido/planner/ompl/Planner.hpp"
+#include "aikido/statespace/GeodesicInterpolator.hpp"
+
+namespace aikido {
+namespace planner {
+namespace ompl {
+
+using aikido::constraint::dart::createProjectableBounds;
+using aikido::constraint::dart::createSampleableBounds;
+using aikido::constraint::dart::createTestableBounds;
+using aikido::constraint::dart::createProjectableBounds;
+using aikido::distance::createDistanceMetric;
+using aikido::statespace::dart::MetaSkeletonStateSpace;
+
+//==============================================================================
+template <class PlannerType>
+OMPLConfigurationToConfigurationPlanner<PlannerType>::
+    OMPLConfigurationToConfigurationPlanner(
+        statespace::ConstStateSpacePtr stateSpace,
+        common::RNG* rng,
+        statespace::ConstInterpolatorPtr interpolator,
+        distance::DistanceMetricPtr dmetric,
+        constraint::SampleablePtr sampler,
+        constraint::TestablePtr boundsConstraint,
+        constraint::ProjectablePtr boundsProjector,
+        double maxDistanceBtwValidityChecks)
+  : ConfigurationToConfigurationPlanner(std::move(stateSpace), rng)
+  , mInterpolator(std::move(interpolator))
```

Comment:
Why do we move the `interpolator` and `boundsConstraint` here, but not the `dmetric`, `sampler`, and `boundsProjector`? I don't have a particular opinion one way or another, but I think it should at least be consistent.

Reply:
Actually, maybe the root question is why are `interpolator` and `boundsConstraint` the only members? One argument seems that the OMPL planner includes references to all of these, but doesn't the OMPL planner use an interpolator somewhere?